lib/utils/tunnel.ts

```diff
@@ -114,12 +118,13 @@ const openPortThroughProxy = (
 				}
 			};
 
+			proxyTunnel.on('data', proxyDataHandler);
 			proxyTunnel.on('timeout', () => {
```

Comment:
I know it was existing code, but it "feels uncomfortable" that the timeout handler is added inside the callback that gets called after a connection is established or throws an error - e.g. after it has timed out. Though I guess event handlers are still called even if added _after_ the event was produced. Would it be clearer to move both the `'timeout'` and `'data'` handlers outside the callback, alongside the `'error'` handler, so that the event handlers are added before the events are produced?

Reply:
It's done this way because we need different timeouts based on the stage of the connection. It takes time for the connect proxy to establish the link and so the timeout is higher before the initial data is sent, then we set it to a normal value.

Reply:
OK. The `setTimeout()` calls would stay where they are, just moving the `on('timeout', ...)` handler setup. But if it works, it works. :-)